common/rpc/peer_chooser.go

```diff
@@ -41,7 +41,7 @@ type (
 	}
 )
 
-func NewDNSPeerChooserFactory(interval time.Duration, logger log.Logger) *dnsPeerChooserFactory {
+func NewDNSPeerChooserFactory(interval time.Duration, logger log.Logger) PeerChooserFactory {
```

Comment:
I'm guessing it complained about "private types are annoying to use"?

Any chance this func is actually only used in this package, and can be made private?

Reply:
NewDNSPeerChooserFactory() is used by cmd/server/cadence/server.go, so no :(

Yeah I agree about returning a specific type instead of the interface being preferable. Better to ignore the lint warning? Does this code even need an interface that has one impl?

Reply:
the repo as a whole is fairly interface-happy / java-flavored, so in a lot of cases I think the answer is no - no need for an interface.

........ but Go makes that kinda hard with tests.
which is probably more reason to reduce our mocks overall and run more "real" code though :|